source/elements/oneDAL/source/algorithms/nearest_neighbors/knn_classification.rst

```diff
@@ -1,3 +1,198 @@
-==================
-kNN classification
-==================
+.. highlight:: cpp
+.. default-domain:: cpp
+
+=========================================
+k-Nearest Neighbors Classification (k-NN)
+=========================================
+
+:math:`k`-NN classification algorithm infers the class for the new feature
+vector by computing majority vote of the :math:`k` nearest observations from the
+training set.
+
+.. |t_math| replace:: `Training <t_math_>`_
+.. |t_api| replace:: `API <t_api_>`_
+.. |t_brute_f| replace:: `Brute-force <t_math_brute_force_>`_
+.. |t_kd_tree| replace:: `k-d tree <t_math_kd_tree_>`_
+.. |t_input| replace:: `train_input <t_api_input_>`_
+.. |t_result| replace:: `train_result <t_api_result_>`_
+.. |t_op| replace:: `train(...) <t_api_>`_
+
+.. |i_math| replace:: `Inference <i_math_>`_
+.. |i_api| replace:: `API <i_api_>`_
+.. |i_brute_f| replace:: `Brute-force <i_math_brute_force_>`_
+.. |i_kd_tree| replace:: `k-d tree <i_math_kd_tree_>`_
+.. |i_input| replace:: `infer_input <i_api_input_>`_
+.. |i_result| replace:: `infer_result <i_api_result_>`_
+.. |i_op| replace:: `infer(...) <i_api_>`_
+
+
+=============== ============= ============= ======== =========== ============
+ **Operation**  **Computational methods**     **Programming Interface**
+--------------- --------------------------- ---------------------------------
+   |t_math|      |t_brute_f|   |t_kd_tree|   |t_op|   |t_input|   |t_result|
+   |i_math|      |i_brute_f|   |i_kd_tree|   |i_op|   |i_input|   |i_result|
+=============== ============= ============= ======== =========== ============
+
+------------------------
+Mathematical formulation
```

Comment:
Minor inconsistency: other algorithms do not have `Mathematical formulation` as a separate section. Are we introducing this?

Reply:
I'd like to change the other algos to align them with kNN.